modules/gateway.go

```diff
@@ -1,12 +1,48 @@
 package modules
 
 import (
+	"io"
+	"net"
+
 	"github.com/NebulousLabs/Sia/consensus"
-	"github.com/NebulousLabs/Sia/network"
 )
 
+// A NetAddress contains the information needed to contact a peer.
+type NetAddress string
+
+// Host returns the NetAddress' IP.
+func (na NetAddress) Host() string {
+	host, _, _ := net.SplitHostPort(string(na))
+	return host
+}
+
+// Port returns the NetAddress' port number.
+func (na NetAddress) Port() string {
+	_, port, _ := net.SplitHostPort(string(na))
+	return port
+}
+
+// A NetConn is a monitored network connection.
+type NetConn interface {
+	io.ReadWriteCloser
+
+	// ReadObject reads and decodes an object from the NetConn. It takes a
+	// maximum length, which the encoded object must not exceed.
+	ReadObject(interface{}, uint64) error
+
+	// WriteObject encodes an object and writes it to the connection.
+	WriteObject(interface{}) error
+
+	// Addr returns the NetAddress of the remote end of the connection.
+	Addr() NetAddress
+}
+
+// RPCFunc is the type signature of functions that handle incoming RPCs.
+type RPCFunc func(NetConn) error
+
 type GatewayInfo struct {
-	Peers []network.Address
+	Address NetAddress
+	Peers   []NetAddress
 }
```

Comment:
I think my comment is out of scope for this PR, but I do believe it would be cleaner to just throw a `Peers` function and an `Address` function into the interface. Info structs just seem dirty.